account_cutoff_prepaid/models/account_cutoff.py

```diff
@@ -0,0 +1,180 @@
+        )]
+
+    @api.multi
+    @api.constrains('start_date', 'end_date', 'forecast')
+    def _check_start_end_dates(self):
+        for prepaid in self:
+            if prepaid.forecast and prepaid.start_date and prepaid.end_date \
+                    and prepaid.start_date > prepaid.end_date:
+                raise ValidationError(_(
+                    'The start date is after the end date!'))
+
+    @api.onchange('forecast')
+    def forecast_onchange(self):
+        return {'warning': {
+            'title': _('Warning'),
+            'message': _(
+                "Don't forget to Re-Generate Lines after entering or "
+                "leaving forecast mode.")}}
+
+    @api.multi
+    def _prepare_prepaid_lines(self, aml, mapping):
+        self.ensure_one()
+        start_date_dt = fields.Date.from_string(aml.start_date)
+        end_date_dt = fields.Date.from_string(aml.end_date)
+        # Here, we compute the amount of the cutoff
+        # That's the important part !
+        total_days = (end_date_dt - start_date_dt).days + 1
+        if self.forecast:
+            out_days = 0
+            forecast_start_date_dt = fields.Date.from_string(self.start_date)
+            forecast_end_date_dt = fields.Date.from_string(self.end_date)
+            if end_date_dt > forecast_end_date_dt:
+                out_days += (end_date_dt - forecast_end_date_dt).days
+            if start_date_dt < forecast_start_date_dt:
+                out_days += (forecast_start_date_dt - start_date_dt).days
+            prepaid_days = total_days - out_days
+        else:
+            cutoff_date_str = self.cutoff_date
+            cutoff_date_dt = fields.Date.from_string(cutoff_date_str)
+            if start_date_dt > cutoff_date_dt:
+                prepaid_days = total_days
+            else:
+                prepaid_days = (end_date_dt - cutoff_date_dt).days
+        assert total_days > 0,\
+            'Should never happen. Total days should always be > 0'
+        cutoff_amount = (aml.debit - aml.credit) *\
+            prepaid_days / float(total_days)
+        # we use account mapping here
+        if aml.account_id.id in mapping:
+            cutoff_account_id = mapping[aml.account_id.id]
+        else:
+            cutoff_account_id = aml.account_id.id
+
+        res = {
+            'parent_id': self.id,
+            'move_line_id': aml.id,
+            'partner_id': aml.partner_id.id or False,
+            'name': aml.name,
+            'start_date': start_date_dt,
+            'end_date': end_date_dt,
+            'account_id': aml.account_id.id,
+            'cutoff_account_id': cutoff_account_id,
+            'analytic_account_id': aml.analytic_account_id.id or False,
+            'total_days': total_days,
+            'prepaid_days': prepaid_days,
+            'amount': aml.credit - aml.debit,
+            'currency_id': self.company_currency_id.id,
+            'cutoff_amount': cutoff_amount,
+            }
```

Comment:
parenthesis one tab back (same for all those occurrences where the closing one does not match visually the opening one)